src/main/java/pro/trevor/tankgame/state/attribute/AttributeObject.java

```diff
@@ -0,0 +1,113 @@
+package pro.trevor.tankgame.state.attribute;
+
+import org.json.JSONObject;
+import pro.trevor.tankgame.util.IJsonObject;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public abstract class AttributeObject {
+
+    protected final Map<String, Object> attributes;
+
+    public AttributeObject() {
+        this.attributes = new HashMap<>();
+    }
+
+    public AttributeObject(Map<String, Object> defaults) {
+        this();
+        for (String attribute : defaults.keySet()) {
+            attributes.put(attribute, defaults.get(attribute));
+        }
+    }
+
+    public AttributeObject(JSONObject json) {
```

Comment:
It would be a good (targetedish) test to verify that an attribute object can be encoded and decoded from a json object and results in the same object.  I say targetedish because it would cover a big chunk of code but it would be a smaller section of code than debug mode,

Reply:
Done